src/components/node/NodeInitializationPopup.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import Button from '../common/Button';
+import Input from '../common/Input';
+import Checkbox from '../common/Checkbox';
+import { PopupButtons, SuccessMessage, ErrorMessage } from '../../styles/PopupStyles';
+import { NodeInitializationResult } from '../../hooks/useNodeManagement';
+
+interface NodeInitializationPopupProps {
+  onInitialize: (nodeName: string, serverPort: number, swarmPort: number, runOnStartup: boolean) => Promise<NodeInitializationResult>;
+  onClose: () => void;
+}
+
+const NodeInitializationPopup: React.FC<NodeInitializationPopupProps> = ({
+  onInitialize,
+  onClose,
+}) => {
+  const [nodeName, setNodeName] = useState('');
+  const [serverPort, setServerPort] = useState('');
+  const [swarmPort, setSwarmPort] = useState('');
+  const [runOnStartup, setRunOnStartup] = useState(false);
+  const [message, setMessage] = useState('');
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const response = await onInitialize(nodeName, parseInt(serverPort), parseInt(swarmPort), runOnStartup);
+    if (response.success) {
+      setMessage(response.message);
+    }else {
```

Comment:
```suggestion
    } else {
```
Parse with something , prettier or lint